core/src/main/java/org/elasticsearch/action/DocWriteResponse.java

```diff
@@ -292,8 +292,8 @@ protected static void declareParserFields(ConstructingObjectParser<? extends Doc
         objParser.declareString(constructorArg(), new ParseField(_ID));
         objParser.declareLong(constructorArg(), new ParseField(_VERSION));
         objParser.declareString(constructorArg(), new ParseField(RESULT));
+        objParser.declareObject(constructorArg(), (p, c) -> ShardInfo.fromXContent(p), new ParseField(_SHARDS));
```

Comment:
No - I changed it to `optionalConstructorArg()` since it can be an argument of the `UpdateResponse`.